Mathlib/Topology/MetricSpace/Pseudo/Constructions.lean

```diff
@@ -235,3 +235,58 @@ protected lemma Filter.Tendsto.nndist {f g : β → α} {x : Filter β} {a b : 
     (hf : Tendsto f x (𝓝 a)) (hg : Tendsto g x (𝓝 b)) :
     Tendsto (fun x => nndist (f x) (g x)) x (𝓝 (nndist a b)) :=
   (continuous_nndist.tendsto (a, b)).comp (hf.prodMk_nhds hg)
+
+section Discrete
+
+namespace PseudoMetricSpace
+
+variable {X : Type*}
+
+/-- The trivial metric space, where every distinct element is 1 away from another.
+This takes an explicit `DiscreteTopology` instance to ensure that the forgetful
+inheritance to topology matches. -/
```

Comment:
Why do we need this?

Reply:
Because I don't want people installing this discrete metric space unless they really mean to.